Python/Product/Common/Infrastructure/JsonRpcWrapper.cs

```diff
@@ -0,0 +1,60 @@
+﻿// Python Tools for Visual Studio
+// Copyright(c) Microsoft Corporation
+// All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the License); you may not use
+// this file except in compliance with the License. You may obtain a copy of the
+// License at http://www.apache.org/licenses/LICENSE-2.0
+//
+// THIS CODE IS PROVIDED ON AN  *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS
+// OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY
+// IMPLIED WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
+// MERCHANTABILITY OR NON-INFRINGEMENT.
+//
+// See the Apache Version 2.0 License for specific language governing
+// permissions and limitations under the License.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.PythonTools.Infrastructure;
+using StreamJsonRpc;
+
+namespace Microsoft.PythonTools.Common.Infrastructure {
+    internal class JsonRpcWrapper {
+
+        private readonly JsonRpc _rpc;
+
+        // list of exception types to ignore
+        private static readonly Type[] _exceptionsToIgnore = {
+            typeof(ObjectDisposedException),
+            typeof(ConnectionLostException)
+        };
+
+        public JsonRpcWrapper(JsonRpc rpc) {
+            _rpc = rpc;
+        }
+
+        public Task NotifyWithParameterObjectAsync(string targetName, object argument = null) {
+
+            if (_rpc is null) {
```

Comment:
Given that this is `readonly`, it seems that this check would never actually be hit?

Reply:
Well since it's a timing issue, it's possible that the _rpc was null when the wrapper was created, and null was passed into the constructor. So that's why I do the check